Add FAQ page (#3324)
microsoft/onnxruntime: docs/ONNX_Runtime_Perf_Tuning.md

# ONNX Runtime Performance Tuning

> ONNX Runtime provides high performance and flexibility of hardware options with the concept of "Execution Providers" to represent different execution kernels. See: [design overview](./HighLevelDesign.md), [supported execution providers](../README.md#supported-accelerators).

Reviewer: ```suggestion
ONNX Runtime gives high performance across a range of hardware options by providing "Execution Providers" to interface to different execution environments. See: [design overview](./HighLevelDesign.md), [supported execution providers](../README.md#supported-accelerators).
```

Along with this flexibility comes decisions for tuning and usage options. For different models and different hardware, there is no silver bullet that can always perform the best. Even for a single execution provider, often there are several knobs that can be tuned (e.g. thread number, wait policy, etc).

This document covers basic tools and knobs that can be leveraged to find the best performance for your model and hardware.

**Topics**
* [Performance Tuning Tools](#Performance-Tuning-Tools)
* [Using different Execution Providers](#Using-different-Execution-Providers)
* [Which Execution Provider will provide the best performance?](#Which-Execution-Provider-will-provide-the-best-performance)
* [Tuning performance for specific Execution Providers](#Tuning-performance-for-specific-Execution-Providers)
* [Troubleshooting model performance issues](#Troubleshooting-model-performance-issues)
***

## Performance Tuning Tools

> The [ONNX Go Live "OLive" tool](https://github.com/microsoft/OLive) is designed as an easy-to-use pipeline for converting models to ONNX and optimizing performance with ONNX Runtime. The tool can help identify the optimal runtime configuration to get the best performance on the target hardware for the model.

Reviewer: ```suggestion
The [ONNX Go Live "OLive" tool](https://github.com/microsoft/OLive) is an easy-to-use pipeline for converting models to ONNX and optimizing performance with ONNX Runtime. The tool can help identify the optimal runtime configuration to get the best performance on the target hardware for the model.
```

  * ACTIVE will not yield CPU, instead it will have a while loop to check whether the next task is ready
  * Use PASSIVE if your CPU usage already high, and use ACTIVE when you want to trade CPU with latency

## Troubleshooting model performance issues

> The answers below are basic troubleshooting suggestions based on common previous user-filed issues and questions. This list is by no means exhaustive and there is a lot of case-by-case fluctuation depending on the model and specific usage scenario. Please use this information to guide your troubleshooting, search through previously filed issues for related topics, and/or file a new issue if your problem is still not resolved.

Reviewer: ```suggestion
The answers below are troubleshooting suggestions based on common previous user-filed issues and questions. This list is by no means exhaustive and there is a lot of case-by-case fluctuation depending on the model and specific usage scenario. Please use this information to guide your troubleshooting, search through previously filed issues for related topics, and/or file a new issue if your problem is still not resolved.
```
